tfjs-backend-webgpu/src/kernels/conv2d_mm_webgpu.ts

```diff
@@ -44,24 +41,22 @@ export class Conv2DMMProgram implements WebGPUProgram {
     util.assert(
         convInfo.dilationHeight === 1 && convInfo.dilationWidth === 1,
         () => 'TODO: Dilation is unimplemented');
-
+    this.dispatchLayout = {x: [1, 2], y: [3], z: [0]};
+    this.workGroupSize =
+        computeWorkGroupSizeForConv2d(this.dispatchLayout, this.outputShape);
     let elementsPerThread: [number, number, number];
     let matMulSource: string;
     if (workPerThread === 0) {
       elementsPerThread = [1, 1, 1];
       matMulSource = makeMatMulSource();
     } else {
-      elementsPerThread = [workPerThread, workPerThread, 1];
-      matMulSource = makeMatMulPackedSource(workPerThread);
+      elementsPerThread =
+          computeWorkPerThreadForConv2d(this.dispatchLayout, this.outputShape);
+      matMulSource = makeMatMulPackedSource(elementsPerThread);
```

Comment:
can't you just pass in this.workGroupSize here, rather than computing elementsPerThread?